modules/market/src/main/java/com/opengamma/strata/market/curve/definition/FxSwapCurveNode.java

```diff
@@ -89,9 +92,10 @@ public DatedCurveParameterMetadata metadata(LocalDate valuationDate) {
 
   @Override
   public FxSwapTrade trade(LocalDate valuationDate, MarketData marketData) {
-    double fxNearRate = marketData.getValue(fxNearKey);
+    FxRate fxNearRate = marketData.getValue(fxNearKey);
+    double rate = fxNearRate.fxRate(fxNearKey.getPair());
```

Comment:
`FxRate` doesn't have a right and a wrong way. You have to pass in the currencies when you request the `double` rate in order to avoid any ambiguity.

It's designed that way to avoid getting in a terrible mess with scenarios. If there were two different rate objects representing the rate for a pair and its inverse then it would be very difficult to keep track. So there is only ever one rate object for a currency pair.